src/overlays/actors/ovl_En_Reeba/z_en_reeba.c

```diff
@@ -21,45 +40,617 @@ const ActorInit En_Reeba_InitVars = {
+    } else {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIVA_APPEAR);
+    }
+
+    this->actionfunc = func_80AE5054;
+}
+
+void func_80AE5054(EnReeba* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 playerLinearVel;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+
+    if ((globalCtx->gameplayFrames & 3) == 0) {
+        func_80033260(globalCtx, &this->actor, &this->actor.posRot.pos, this->actor.shape.unk_10, 1, 8.0f, 0x1F4, 0xA,
+                      1);
+    }
+
+    if (this->unk_278 == 0) {
+        Math_SmoothScaleMaxF(&this->actor.shape.unk_10, 12.0f, 1.0f, 1.0f);
+        if (this->actor.shape.unk_08 < 0.0f) {
+            Math_SmoothDownscaleMaxF(&this->actor.shape.unk_08, 1.0f, this->unk_288);
+            Math_SmoothScaleMaxF(&this->unk_288, 300.0f, 1.0f, 5.0f);
+
+        } else {
+            this->unk_288 = 0.0f;
+            this->actor.shape.unk_08 = 0.0f;
+            playerLinearVel = player->linearVelocity;
+
+            switch (this->unk_280) {
+                case 0:
+                    this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+                    break;
+                case 1:
+                    this->actor.posRot.rot.y = this->actor.yawTowardsLink + (800.0f * playerLinearVel);
+                    break;
+                case 2:
+                case 3:
+                    this->actor.posRot.rot.y =
+                        this->actor.yawTowardsLink +
+                        (player->actor.shape.rot.y - this->actor.yawTowardsLink) * (playerLinearVel * 0.15f);
+                    break;
+                case 4:
+                    this->actor.posRot.rot.y = this->actor.yawTowardsLink - (800.0f * playerLinearVel);
+            }
+
+            if (this->isBig) {
+                this->actionfunc = func_80AE538C;
+            } else {
+                this->unk_272 = 130;
+                this->actor.speedXZ = Math_Rand_ZeroFloat(4.0f) + 6.0f;
+                this->actionfunc = func_80AE5270;
+            }
+        }
+    }
+}
+
+void func_80AE5270(EnReeba* this, GlobalContext* globalCtx) {
+    s32 surfaceType;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+
+    if (this->actor.shape.unk_10 < 12.0f) {
+        Math_SmoothScaleMaxF(&this->actor.shape.unk_10, 12.0f, 3.0f, 1.0f);
+    }
+
+    surfaceType = func_80041D4C(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorPolySource);
+
+    if ((surfaceType != 4) && (surfaceType != 7)) {
+        this->actionfunc = func_80AE5688;
+        this->actor.speedXZ = 0.0f;
+    } else {
+        if ((this->unk_272 == 0) || (this->actor.xzDistFromLink < 30.0f) || (this->actor.xzDistFromLink > 400.0f) ||
+            (this->actor.bgCheckFlags & 8)) {
+            this->actionfunc = func_80AE5688;
+        } else if (this->unk_274 == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIVA_MOVE);
+            this->unk_274 = 10;
+        }
+    }
```

Comment:
```suggestion
    } else if ((this->unk_272 == 0) || (this->actor.xzDistFromLink < 30.0f) || (this->actor.xzDistFromLink > 400.0f) ||
        (this->actor.bgCheckFlags & 8)) {
        this->actionfunc = func_80AE5688;
    } else if (this->unk_274 == 0) {
        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIVA_MOVE);
        this->unk_274 = 10;
    }
```

Reply:
imo the way i have it already makes more sense
```
if (surafce types conditions are met) {
    do stuff
} else {
    do other stuff opposite of surface type stuff
}
```
maybe this will change when stuff is named and the conditions/functions being set will look more related to each other, but yea this is my imo for now

Reply:
I think I agree with krim here. Or maybe they should even be early returns since they are technically transitions to other actions, so I think it would make sense to say the execution for the current action stops at that point.